src/odemis/acq/stream/_sync.py

```diff
@@ -1783,7 +1783,12 @@ def _onCompletedData(self, n, raw_das):
         for d in raw_das:
             d.metadata[MD_DESCRIPTION] = sname
 
-        if len(raw_das) != numpy.prod(self.repetition.value):
+        # if polarization analyzer HW: can be 6 images per beam pos
+        if hasattr(self._streams[1], "analyzer") and self._streams[1].acquireAllPol.value == True:
```

Comment:
Use `analyzer` and `acquireAllPol` as in the rest of the code (`._runAcquisitionEbeam()`): directly on `self`. You can also skip the `== True`, and just write `self._acquireAllPol.value`.

I'd also suggest to compute the "expected number of images" explicitly. For instance start with `exp_num_image = numpy.prod(self.repetition.value)`, and multiply it if needed. That would have the advantage of showing the right number in the log message too.